azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -551,12 +559,29 @@ public Object getMBeanAttribute(ObjectName name, String attribute) {
    *
    * @return hostname
    */
-  public String getHost() {
-    String host = "unkownHost";
+  public static String getHost() {
+    if(StringUtils.isEmpty(hostAlias)) {
+      URL scriptUrl = Resources.getResource(ServerInternals.HOST_NAME_ALIAS_SCRIPT);
+      if (scriptUrl != null) {
+        try {
+          File aliasFile = new File(scriptUrl.toURI());
+          hostAlias = FileIOUtils.runShellCommand("perl " + aliasFile, new File(".")).trim();
```

Comment:
Questions:
 - What are the advantages of using perl to do this vs directly doing this in Java?
 - probably should be abstracted in a separate method / class since there's a lot going on to determine the vip?

Some comments on what's going on here might help.